apps/envited-x-data-space/app/api/tokens/[fullResourceName]/route.ts

```diff
@@ -0,0 +1,40 @@
+import { NextResponse } from 'next/server'
+import { isEmpty, isNil, prop } from 'ramda'
+
+import { getServerSession } from '../../../../common/auth'
+import { ERRORS, ERROR_CODES } from '../../../../common/constants/errors'
+import { db } from '../../../../common/database/queries'
+import { parseGlobalIdentifier } from '../../../../common/globalIdentifiers'
+
+export async function GET(
+  request: Request,
+  { params: { fullResourceName } }: { params: { fullResourceName: string } },
+) {
+  try {
+    if (isNil(fullResourceName) || isEmpty(fullResourceName)) {
+      return NextResponse.json({ error: ERRORS.GLOBAL_IDENTIFIER_MISSING }, { status: ERROR_CODES.BAD_REQUEST })
+    }
+
+    const session = await getServerSession()
+
+    if (isNil(session)) {
+      return NextResponse.json({ error: ERRORS.UNAUTHORIZED }, { status: ERROR_CODES.UNAUTHORIZED })
+    }
+
+    const { scopedIdentifier } = parseGlobalIdentifier(fullResourceName)
+
+    const connection = await db()
+    const globalIdentifier = await connection.getGlobalIdentifierByScopedIdentifier(scopedIdentifier)
+
+    if (isNil(globalIdentifier)) {
+      return NextResponse.json(null)
+    }
+
+    const token = await connection.getTokenByWebGlobalIdentifierId(prop('id')(globalIdentifier))
```

Comment:
Maybe add a guard if the retrieved global identifier is a did:web method. If not send back a bad request. That way this function can be expanded with multiple methods potentially. Right now it doesn't differ much from sending "just" the scoped identifier.